### ISSUE TITLE
Support for S3:Events Prefix and Suffix Rules

### ISSUE THREAD

**Issue by malcyL** [ar90n/serverless-s3-local#27]
Serverless allows for the setting of prefix and suffix rules on S3 Events:

https://serverless.com/framework/docs/providers/aws/events/s3#setting-filter-rules

```
functions:
  users:
    handler: incoming.handler
    events:
      - s3:
          bucket: mybucket
          event: s3:ObjectCreated:*
          rules:
            - prefix: incoming/
            - suffix: .zip
```

I think serverless-s3-local is ignoring these rules?

My use case is an existing system which changes the key of a file after it's completed processing. For example, a file is uploaded to s3 with a bucket and key: ```s3://mybucket/incoming/file1.zip``` and then after processing it is moved to ```s3://mybucket/processed/file1.zip```.

When running locally with serverless-s3-local the handler is triggered twice, once when it is initially created and again when the key is changed. When deployed to AWS this doesn't happen as the prefix and suffix rules are applied.

**Comment by ar90n** [comment]
Hi @malcyL 
Thanks for your report.
Now, serverless-s3-local doesn't support these rules.
But I decided to try to support this function.
Please wait for a while.

**Comment by malcyL** [comment]
Hi @ar90n 

I've tested your new release this morning and it's working as expected.

The lambda is triggered when the file is first created and the key matches the rules. After processing when we change the key of the file, the lambda is no longer triggered because the new key doesn't match the rules.

I've also tested with multiple lambda's all working with different prefixes. i.e. Lambda 1 processing files from `s3://mybucket/images/incoming` and Lambda 2 processing files from `s3://mybucket/data/incoming`. Previously, incoming files would trigger both the lambda's. Now only the correct lambda is triggered. 

I'm happy for this issue to be closed, but I'll leave it for you to do.

Thank you so much for turning this feature request around so quickly. That's very impressive.
